src/transport/zerotier/zerotier.c

```diff
@@ -845,7 +845,7 @@ zt_pipe_recv_data(zt_pipe *p, const uint8_t *data, size_t len)
 	// that we only can catch the case where a message is larger by
 	// more than a fragment, since the final fragment may be shorter,
 	// and we won't know that until we receive it.
-	if ((nfrags * fragsz) >= (p->zp_rcvmax + fragsz)) {
+	if ((nfrags * fragsz) >= ((uint16_t)(p->zp_rcvmax) + fragsz)) {
```

Comment:
Yeah, pretty sure uint16_t is too small here.  This needs to compare against uintptr_t or something like that.  (I suspect size_t is right, actually.)

Or (and I think this is better) -- cast the *other* side to int.  I think we want to cast "up" rather than down.

Reply:
Thanks! Yes, I found out now that nothing was coming through. ;-)